iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/helplets.inl

```diff
@@ -48,34 +82,32 @@ inline bool isValidFileName(const string<StringCapacity>& name) noexcept
     }
 
     // check if the file contains only valid characters
-    for (uint64_t i = 0; i < nameSize; ++i)
-    {
-        const char c = name.c_str()[i];
-        if (!((internal::ASCII_A <= c && c <= internal::ASCII_Z)
-              || (internal::ASCII_CAPITAL_A <= c && c <= internal::ASCII_CAPITAL_Z)
-              || (internal::ASCII_0 <= c && c <= internal::ASCII_9) || c == internal::ASCII_MINUS
-              || c == internal::ASCII_DOT || c == internal::ASCII_COLON || c == internal::ASCII_UNDERSCORE))
-        {
-            return false;
-        }
-    }
-
-    return true;
+    return isValidPathEntry(name);
 }
 
 template <uint64_t StringCapacity>
-inline bool isValidFilePath(const string<StringCapacity>& name) noexcept
+inline bool isValidPathToFile(const string<StringCapacity>& name) noexcept
 {
     if (doesEndWithPathSeparator(name))
     {
         return false;
     }
 
-    return isValidPath(name);
+    auto lastSeparatorPosition = name.find_last_of(platform::IOX_PATH_SEPARATORS);
+    auto filePart = (lastSeparatorPosition) ? name.substr(*lastSeparatorPosition + 1).value() : name;
+    auto pathPart = (lastSeparatorPosition) ? name.substr(0, *lastSeparatorPosition).value() : string<StringCapacity>();
+
+    return (pathPart.empty() || isValidPathToDirectory(pathPart)) && isValidFileName(filePart);
 }
 
 template <uint64_t StringCapacity>
-inline bool isValidPath(const string<StringCapacity>& name) noexcept
+inline bool isValidFilePath(const string<StringCapacity>& name) noexcept
+{
+    return isValidPathToFile(name);
+}
```

Comment:
This neither

Reply:
You are right but this was part of the public API of iceoryx 2.0. Therefore I thought to keep this function available and deprecate it for now. Can be deleted in iceoryx 4.0.

Reply:
Yes, we should at least deprecate it. I'm not sure what guarantees we currently give for non-posh related API. In the end, the change from the creation pattern to the builder pattern of the FileLock is also a breaking change and there is no migration path via a deprecated API but just the new API. Therefore I would also slightly lean towards just removing this functions since they are not used in our public API.

Reply:
I removed it